test/AutoDiff/protocol_requirement_autodiff.swift

```diff
@@ -1,4 +1,4 @@
-// RUN: %target-run-simple-swift
+// RUN: %target-swift-frontend -typecheck -verify %s
```

Comment:
We should not change this here. This was a runtime test suite, and `-typecheck` disabled all runtime tests in this file.

All type checking verification tests should be added to `test/AutoDiff/differentiable_attr_type_checking.swift`.